staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,598 @@
+	cmd.Flags().StringVar(&flags.Role, "role", flags.Role, "Set a label selector by node role to filter on.")
+	cmd.Flags().StringVarP(&flags.Selector, "selector", "l", flags.Selector, "Selector (label query) to filter on.")
+	cmd.Flags().BoolVar(&flags.Raw, "raw", flags.Raw, "Perform no transformation of the returned data.")
+	cmd.Flags().BoolVar(&flags.Unify, "unify", flags.Unify, "Interleave logs by sorting the output. Defaults on when viewing node journal logs.")
+}
+
+// ToOptions converts from CLI inputs to runtime inputs.
+func (flags *NodeLogsCmdFlags) ToOptions(args []string) (*NodeLogsCmdOptions, error) {
+	o := &NodeLogsCmdOptions{
+		Boot:                         &flags.Boot,
+		Path:                         flags.Path,
+		InsecureSkipTLSVerifyBackend: flags.InsecureSkipTLSVerifyBackend,
+		RESTClientGetter:             flags.RESTClientGetter,
+		TailLines:                    &flags.TailLines,
+		PatternCaseSensitive:         flags.PatternCaseSensitive,
+		LogsForObject:                polymorphichelpers.LogsForObjectFn,
+		Resources:                    args,
+	}
+
+	o.Unify = flags.Path == "journal" || flags.Unify
+
+	o.Raw = flags.Raw || flags.Path == "journal"
+
+	if len(flags.Services) > 0 {
+		o.Services = flags.Services
+	}
+
+	if len(flags.SinceTime) > 0 {
+		t, err := util.ParseRFC3339(flags.SinceTime, metav1.Now)
+		if err != nil {
+			return nil, err
+		}
+
+		o.SinceTime = &t
+	}
+
+	if len(flags.UntilTime) > 0 {
+		t, err := util.ParseRFC3339(flags.UntilTime, metav1.Now)
+		if err != nil {
+			return nil, err
+		}
+
+		o.UntilTime = &t
+	}
+
+	if len(flags.Output) > 0 {
+		o.Format = flags.Output
+	}
+
+	if len(flags.Pattern) > 0 {
+		o.Pattern = flags.Pattern
+	}
+
+	if len(flags.Selector) > 0 && o.TailLines == nil {
+		o.TailLines = &nodeSelectorTail
+	}
+
+	if o.Object == nil {
+		builder := resource.NewBuilder(flags.RESTClientGetter).
+			WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+			SingleResourceType()
+		if len(args) > 0 {
+			builder.ResourceNames("nodes", args...)
+		}
+		if len(flags.Role) > 0 {
+			req, err := labels.NewRequirement(fmt.Sprintf("node-role.kubernetes.io/%s", flags.Role), selection.Exists, nil)
+			if err != nil {
+				return nil, fmt.Errorf("invalid --role: %v", err)
+			}
+			flags.Selector = req.String()
+		}
+		if len(flags.Selector) > 0 {
+			o.Selector = flags.Selector
+			builder.ResourceTypes("nodes").LabelSelectorParam(o.Selector)
+		}
+		result := builder.Do()
+
+		o.Prefix = o.Prefix || !result.TargetsSingleItems()
+
+		infos, err := result.Infos()
+		if err != nil {
+			return nil, err
+		}
+
+		if flags.Selector == "" && len(infos) != 1 {
+			return nil, fmt.Errorf("expected a node resource")
+		}
+
+		o.Object = infos[0].Object
+		if flags.Selector != "" && len(o.Object.(*corev1.NodeList).Items) == 0 {
+			return nil, fmt.Errorf("no node resources found")
+		}
+	}
+
+	// validation
```

Comment:
Nit: that bit should probably be part of separate `Validation` method, responsible for ensuring that `NodeLogsCmdOptions` struct is ready for `Run` invocation.

Reply:
added the `Validate` method